Library/Homebrew/cmd/tests.rb

```diff
@@ -36,18 +24,55 @@ def tests
         args << "TESTOPTS=--name=test_#{test_method}" if test_method
       end
       args += ARGV.named.select { |v| v[/^TEST(OPTS)?=/] }
-      system "bundle", "exec", "rake", "test", *args
 
-      Homebrew.failed = !$?.success?
+      begin
+        sanitize_env!
+
+        # Since we "borrow" HOME for duration of tests bundler
+        # needs fetching/installing every run.
+        Homebrew.install_gem_setup_path! "bundler"
+        unless quiet_system("bundle", "check")
+          system "bundle", "install", "--path", "vendor/bundle"
+        end
```

Comment:
Actually, I have an idea: Move this code back before the `begin` and then once the temporary test home has been set up in `sanitize_env!`, create a symlink in that test home that points to the `.gem` subdirectory in the original home directory. (This assumes that the tests themselves won't mess with the `.gem` directory in `ENV["HOME"]`, which seems a reasonable assumption thus far.)